src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateServiceTest.kt

```diff
@@ -1,160 +1,199 @@
   }
 
   @Test
   fun `earliestReleaseDate should return (2018-04-30) if ARD is (2018-05-07) and (2018-05-02) is bank holiday`() {
     val actualReleaseDate = LocalDate.parse("2018-05-07")
 
-    val earliestPossibleReleaseDate = service.getEarliestReleaseDate(actualReleaseDate)
+    val earliestPossibleReleaseDate = getEarliestDate(actualReleaseDate)
     assertThat(earliestPossibleReleaseDate).isEqualTo(LocalDate.parse("2018-04-30"))
   }
 
   @Test
   fun `earliestReleaseDate should return (2018-07-03) if ARD is (2018-07-06) third working day before CRD`() {
     val actualReleaseDate = LocalDate.parse("2018-07-06")
 
-    val earliestPossibleReleaseDate = service.getEarliestReleaseDate(actualReleaseDate)
+    val earliestPossibleReleaseDate = getEarliestDate(actualReleaseDate)
     assertThat(earliestPossibleReleaseDate).isEqualTo(LocalDate.parse("2018-07-03"))
   }
 
   @Test
   fun `earliestReleaseDate should return (2018-07-03) if ARD is (2018-07-07) as it is not a bank holiday or weekend`() {
     val actualReleaseDate = LocalDate.parse("2018-07-07")
 
-    val earliestPossibleReleaseDate = service.getEarliestReleaseDate(actualReleaseDate)
+    val earliestPossibleReleaseDate = getEarliestDate(actualReleaseDate)
     assertThat(earliestPossibleReleaseDate).isEqualTo(LocalDate.parse("2018-07-03"))
   }
 
   @Test
   fun `earliestReleaseDate should return (2018-07-03) if ARD is (2018-07-08) as it is not a bank holiday or weekend`() {
     val actualReleaseDate = LocalDate.parse("2018-07-08")
 
-    val earliestPossibleReleaseDate = service.getEarliestReleaseDate(actualReleaseDate)
+    val earliestPossibleReleaseDate = getEarliestDate(actualReleaseDate)
     assertThat(earliestPossibleReleaseDate).isEqualTo(LocalDate.parse("2018-07-03"))
   }
 
   @Test
   fun `earliestReleaseDate should return (2018-07-03) if CRD is (2018-07-08) as it is not a bank holiday or weekend`() {
     val actualReleaseDate = LocalDate.parse("2018-07-08")
 
-    val earliestPossibleReleaseDate = service.getEarliestReleaseDate(actualReleaseDate)
+    val earliestPossibleReleaseDate = getEarliestDate(actualReleaseDate)
     assertThat(earliestPossibleReleaseDate).isEqualTo(LocalDate.parse("2018-07-03"))
   }
 
+  @Test
+  fun `isLateAllocationWarningRequired should return true if the difference between release date and modified date is equal to 5 days`() {
+    assertTrue(service.isLateAllocationWarningRequired(LocalDate.parse("2023-11-17")))
+  }
+
+  @Test
+  fun `isLateAllocationWarningRequired should return true if the difference between release date and modified date is less than 5 days`() {
+    assertTrue(service.isLateAllocationWarningRequired(LocalDate.parse("2023-11-14")))
+  }
+
+  @Test
+  fun `isLateAllocationWarningRequired should return true if the difference between release date and modified date is more than 5 days`() {
+    assertFalse(service.isLateAllocationWarningRequired(LocalDate.parse("2023-11-20")))
```

Comment:
best to test the boundaries when working on things like this.

Around each end of the range:
``` 
The day before the limit 
The day at the limit
The day after the limit
```

Reply:
Your logic is very specific and your tests should fit closely around it.